File: shared/src/types/recovery-map.ts
import type { BodyRegion, RecoveryGoal, RecoverySignalType } from "./client-profile.js";

export interface RecoveryMap {
  clientId: string;
  highlightedRegions: HighlightedRegion[];
  wearableContext: WearableContext | null;
  priorSessions: PriorSessionSummary[];
  suggestedGoal: RecoveryGoal;
  generatedAt: string;
}

export interface HighlightedRegion {
  region: BodyRegion;
  severity: number;
  signalType: RecoverySignalType;
  romDelta: number | null;
  trend: 'improving' | 'declining' | 'stable' | null;
  asymmetryFlag: boolean;
  compensationHint: string | null;
}

export interface WearableContext {
  hrv: number;
  strain: number;
  sleepScore: number;
  lastSync: string;
}

export interface PriorSessionSummary {
  date: string;
  configSummary: string;
  outcomeRating: number;
}

File: HydraScan/HydraScan/Models/RecoveryMap.swift
import Foundation

struct HighlightedRegion: Identifiable, Codable, Hashable {
    var id = UUID()
    var region: BodyRegion
    var severity: Int
    var signalType: RecoverySignalType
    var romDelta: Double?
    var asymmetryFlag: Bool
    var compensationHint: String?
}

struct WearableContext: Codable, Hashable {
    var hrv: Double?
    var strain: Double?
    var sleepScore: Double?
    var lastSync: Date?
}

struct PriorSessionSummary: Identifiable, Codable, Hashable {
    var id = UUID()
    var date: Date
    var configurationSummary: String
    var outcomeRating: String?
}

struct RecoveryMap: Codable, Hashable {
    var highlightedRegions: [HighlightedRegion]
    var wearableContext: WearableContext?
    var priorSessions: [PriorSessionSummary]
    var suggestedGoal: RecoveryGoal?
    var generatedAt: Date
}
